automock ignores uses statements in module bodies

Great, thank you. That works as is, but extending a little further I'm stuck on the following scoping issue. I can open as a separate issue if that would be better. 

some_module.rs:
```rust
use cfg_if::cfg_if;
#[cfg(test)]
use mockall::automock;

pub struct SomeStruct {}

#[cfg_attr(test, automock)]
pub mod inner {
    use super::SomeStruct;
    pub fn free_function(_arg: SomeStruct) {}
}

cfg_if! {
    if #[cfg(test)] {
        pub use mock_inner::*;
    } else {
        pub use inner::*;
    }
}
```
main.rs:

```rust
#![feature(proc_macro_hygiene)]
#[cfg(test)]
use mockall::predicate::*;

pub mod some_module;
use some_module::*;

fn my_func() {
    let s = some_module::SomeStruct {};
    some_module::free_function(s);
}

fn main() {
}

#[test]
fn some_test() {
    let context = some_module::free_function_context();
    context.expect().times(1).with(always()).returning(|_x| ());
    my_func();
}
```

gives:
```
error[E0412]: cannot find type `SomeStruct` in this scope
  --> src/some_module.rs:10:32
   |
10 |     pub fn free_function(_arg: SomeStruct) {}
   |                                ^^^^^^^^^^ not found in this scope
```

...

It works if some_module.rs is updated to only uses absolute paths:  
```rust
#[cfg_attr(test, automock)]
pub mod inner {
    pub fn free_function(_arg: crate::some_module::SomeStruct) {}
}
```

_Originally posted by @lromeo in https://github.com/asomers/mockall/issues/49#issuecomment-531594357_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,17 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased] - ReleaseDate
+### Added
+### Changed
+### Fixed
+
+- Fixed automocking modules that contain use statements importing types that
+  are used in function signatures.
+  ([#53](https://github.com/asomers/mockall/pull/53))
+
+### Removed
+
 ## [0.5.0] - 5 September 2019
 
 ### Added

--- a/mockall/tests/automock_module.rs
+++ b/mockall/tests/automock_module.rs
@@ -11,13 +11,15 @@ cfg_if! {
     if #[cfg(feature = "nightly")] {
         mod m {
             use mockall::*;
+            type T = u32;
 
             #[automock]
             #[allow(unused)]
             mod foo {
-                pub fn bar(_x: u32) -> i64 {unimplemented!()}
+                use super::T;
+                pub fn bar(_x: T) -> i64 {unimplemented!()}
                 // We must have a separate method for every should_panic test
-                pub fn bar1(_x: u32) -> i64 {unimplemented!()}
+                pub fn bar1(_x: T) -> i64 {unimplemented!()}
             }
 
             #[test]

--- a/mockall_derive/src/automock.rs
+++ b/mockall_derive/src/automock.rs
@@ -551,8 +551,7 @@ fn mock_module(mod_: ItemMod) -> TokenStream {
     };
     for item in items.iter() {
         match item {
-            Item::ExternCrate(_) | Item::Use(_)
-                | Item::Impl(_) =>
+            Item::ExternCrate(_) | Item::Impl(_) =>
             {
                 // Ignore
             },
@@ -584,6 +583,10 @@ fn mock_module(mod_: ItemMod) -> TokenStream {
             Item::TraitAlias(ta) => {
                 // Copy verbatim
                 ta.to_tokens(&mut body)
+            },
+            Item::Use(u) => {
+                // Copy verbatim
+                u.to_tokens(&mut body)
             },
             _ => compile_error(item.span(), "Unsupported item"),
         }

